services/node_manager/src/test/java/com/futurewei/alcor/nodemanager/controller/NodeControllerTest.java

```diff
@@ -0,0 +1,176 @@
+/*Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.nodemanager.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.futurewei.alcor.nodemanager.entity.NodeInfo;
+import com.futurewei.alcor.nodemanager.entity.NodeInfoJson;
+import com.futurewei.alcor.nodemanager.service.NodeService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.net.InetAddress;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class NodeControllerTest {
```

Comment:
Current UTs test the happy path. Should add more UTs to cover the sad path. 

For example, what is supposed to happen if a hacker tries to send some random Ip and Mac addresses to NodeManager? The correct behavior is to reject those calls. 

Don't see necessary verification for NodeManager.

Reply:
Didn't see the response so I am assuming that you are still working on these UTs. 

A good discussion about UT handling on the Alcor channel on Slack today. Please take a look.

Reply:
More UTs for sad pass and exception handling cases.

Reply:
Looks better now. Let us keep looking to see if any more cases might be missed.